tests/Discuss/Repository/DiscussionRepository.php

```diff
@@ -248,4 +248,196 @@ public function test_getAll_PaginationSucceed()
             ->array(iterator_to_array($pager))->hasSize(0)
         ;
     }
+
+    /**
+     * @param array  $repos
+     * @param string $date
+     * @param int    $recId
+     * @param int    $iniId
+     * @param string $recStatus
+     * @param string $iniStatus
+     * @return Message
+     */
+    public function generateDiscussionAndMessage (
+        array $repos,
+        $date,
+        $recId,
+        $iniId,
+        $recStatus,
+        $iniStatus
+    )
+    {
+        $message = $this->createMessage();
+        $message->setSendDate(new \DateTime($date));
+        $message->getDiscussion()
+            ->setRecipient($recId)
+            ->setInitiator($iniId)
+        ;
+        $repos['message']->save($message);
+        $message->getDiscussion()
+            ->setUserStatus($recId, new Status($recStatus))
+            ->setUserStatus($iniId, new Status($iniStatus))
+        ;
+        $repos['message']->save($message);
+        $repos['discuss']->save($message->getDiscussion());
+
+        return $message;
+    }
+
+    public function test_getAllOrderedByMessageSendDate_()
+    {
+        $repos['client'] = new Client();
+        $repos['discuss'] = $repos['client']->getDiscussionRepository();
+        $repos['message'] = $repos['client']->getMessageRepository();
+
+        $message1 = $this->generateDiscussionAndMessage($repos, '10-11-2018', 1111, 1112, Status::HIDDEN, Status::HIDDEN);
+        $message2 = $this->generateDiscussionAndMessage($repos, '10-11-2019', 2221, 2222, Status::DISPLAYED, Status::DISPLAYED);
+        $message3 = $this->generateDiscussionAndMessage($repos, '10-11-2020', 3331, 3332, Status::HIDDEN, Status::DISPLAYED);
+        $message4 = $this->generateDiscussionAndMessage($repos, '10-12-2019', 4441, 4442, Status::DISPLAYED, Status::HIDDEN);
+
+        // Get discussion without filter
+        $result = [];
+        $expected = [
+            $message3,
+            $message3->getDiscussion(),
+            $message4,
+            $message4->getDiscussion(),
+            $message2,
+            $message2->getDiscussion(),
+            $message1,
+            $message1->getDiscussion(),
+        ];
+
+        foreach ($repos['discuss']->getAllOrderedByMessageSendDate() as $d) {
```

Comment:
évites les abbr. il faut que le nom de la variable nous laisse comprendre ce qui lui est assigné.
```suggestion
        foreach ($repos['discuss']->getAllOrderedByMessageSendDate() as $discussion) {
```